internal/controllers/core/cmd/controller.go

```diff
@@ -151,11 +152,17 @@ func (c *Controller) lastStartEventTime(startOn *StartOnSpec, buttons map[string
 			continue
 		}
 		lastEventTime := b.Status.LastClickedAt
-		if !lastEventTime.Time.Before(startOn.StartAfter.Time) && lastEventTime.Time.After(cur) {
-			cur = lastEventTime.Time
+		if !lastEventTime.Time.Before(startOn.StartAfter.Time) && lastEventTime.Time.After(latestTime) {
+			latestTime = lastEventTime.Time
+			latestButton = b
 		}
 	}
-	return cur
+
+	var inputs []v1alpha1.UIInputStatus
+	if latestButton != nil {
+		inputs = latestButton.Status.Inputs
```

Comment:
this seems surprising to me? i would have expected you'd want to grab the inputs from all buttons, rather than the most recently clicked button

Reply:
I don't think I have a strong opinion on this and I think I'm fine going either way until we come up with a concrete use case for a cmd with multiple buttons in its `StartOn`.

I went this way because it seemed like it'd surprise me if I clicked a button and my process got values from other buttons that seem (to me) not really related.

My mental model is a button is like an html `<form>` with a mandatory `input type="submit"`, and if two forms have the same `action`, they don't get each others `input`s.

Or, from another angle - a cmd has some set of env vars it consumes, which we could call the api it exposes to buttons that want to trigger it. If a cmd consumes inputs from all of its buttons then things get messy if two buttons representing two workflows try to set the same env var (or maybe not *that* messy - we just take the most recent one for each).

I'm fine shipping as-is and changing as needed, or changing to match your expectations if you have a strong expectation it should go that way, or chatting. I've been trying to think of use cases that'd tilt it one way or another and am starting to think having one cmd triggered by multiple buttons doesn't currently have a compelling use case (vs declaring multiple cmds, one per button)

Reply:
i buy that argument! ya, I haven't totally thought through what it means to to have multiple buttons with inputs